From 3353bd627b92bd06f9722e475e2ddad4ea9eec48 Mon Sep 17 00:00:00 2001
From: David Daney <david.daney@cavium.com>
Date: Mon, 13 May 2019 14:48:00 +0200
Subject: [PATCH 276/345] MIPS: pcie-octeon:  Use level semantics for int-A
 interrupts.

When interrupts are shared, the CIU gets stuck if interrupts use edge
semantics.

Force level semantics.

Signed-off-by: David Daney <david.daney@cavium.com>
---
 arch/mips/pci/pcie-octeon.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/arch/mips/pci/pcie-octeon.c b/arch/mips/pci/pcie-octeon.c
index 322e1a2cbf86..e1bf78e79617 100644
--- a/arch/mips/pci/pcie-octeon.c
+++ b/arch/mips/pci/pcie-octeon.c
@@ -132,8 +132,10 @@ static int __init octeon_pcie78xx_pcibios_map_irq(const struct pci_dev *dev,
 
 	irq = irq_create_mapping(d, intsn);
 
-	if (!OCTEON_IS_MODEL(OCTEON_CN78XX_PASS1_X))
+	if (!OCTEON_IS_MODEL(OCTEON_CN78XX_PASS1_X)) {
+		irq_set_irq_type(irq, IRQ_TYPE_LEVEL_HIGH);
 		return irq;
+	}
 
 	WARN_ON(pcie->node >= ARRAY_SIZE(pcie_17400_irqs));
 	WARN_ON(pin >= ARRAY_SIZE(pcie_17400_irqs[0]));
@@ -157,7 +159,7 @@ static int __init octeon_pcie78xx_pcibios_map_irq(const struct pci_dev *dev,
 		goto err;
 	}
 
-	irqd_set_trigger_type(irq_get_irq_data(irq), IRQ_TYPE_EDGE_RISING);
+	irq_set_irq_type(irq, IRQ_TYPE_EDGE_RISING);
 
 	irq_set_status_flags(irq, IRQ_NOAUTOEN);
 	rv = request_irq(irq, pcie_17400_handler, IRQF_NO_THREAD, "inta-war", cd);
-- 
2.25.1

